src/commands/images/buildImage.ts

```diff
@@ -20,37 +21,42 @@ export async function buildImage(context: IActionContext, dockerFileUri: vscode.
 
     const dockerFileItem = await quickPickDockerFileItem(context, dockerFileUri, rootFolder);
 
-    let contextPath: string = dockerFileItem.relativeFolderPath;
-    if (defaultContextPath && defaultContextPath !== '') {
-        contextPath = defaultContextPath;
-    }
-    let absFilePath: string = path.join(rootFolder.uri.fsPath, dockerFileItem.relativeFilePath);
-    let dockerFileKey = `buildTag_${absFilePath}`;
-    let prevImageName: string | undefined = ext.context.globalState.get(dockerFileKey);
-    let suggestedImageName: string;
-
-    if (!prevImageName) {
-        // Get imageName based on name of subfolder containing the Dockerfile, or else workspacefolder
-        suggestedImageName = path.basename(dockerFileItem.relativeFolderPath).toLowerCase();
-        if (suggestedImageName === '.') {
-            suggestedImageName = path.basename(rootFolder.uri.fsPath).toLowerCase();
-        }
-
-        suggestedImageName += ":latest"
+    const task = await getOfficialBuildTaskForDockerfile(dockerFileItem.absoluteFilePath, rootFolder);
+    if (task) {
+        await vscode.tasks.executeTask(task);
     } else {
-        suggestedImageName = prevImageName;
-    }
+        let contextPath: string = dockerFileItem.relativeFolderPath;
+        if (defaultContextPath && defaultContextPath !== '') {
+            contextPath = defaultContextPath;
+        }
+        let absFilePath: string = path.join(rootFolder.uri.fsPath, dockerFileItem.relativeFilePath);
+        let dockerFileKey = `buildTag_${absFilePath}`;
+        let prevImageName: string | undefined = ext.context.globalState.get(dockerFileKey);
+        let suggestedImageName: string;
+
+        if (!prevImageName) {
+            // Get imageName based on name of subfolder containing the Dockerfile, or else workspacefolder
+            suggestedImageName = path.basename(dockerFileItem.relativeFolderPath).toLowerCase();
```

Comment:
>basename [](start = 38, length = 8)

DirectoryName.toLower isn't guaranteed to be a valid image name.

Reply:
No, but this isn't code I wrote and nobody's complained in the past.

Reply:
Suggest to consolidate with the code you've written to generate a valid image name